manimlib/mobject/svg/text_mobject.py

```diff
@@ -91,75 +88,14 @@ def __init__(self, text, **config):
-        context.select_font_face(self.font, self.str2slant(self.slant), self.str2weight(self.weight))
-        context.move_to(START_X, START_Y)
-        context.show_text("_")
-        surface.finish()
-        svg_with_space = SVGMobject(file_name, height=self.height,
-                                    width=self.width,
-                                    stroke_width=self.stroke_width,
-                                    should_center=self.should_center,
-                                    unpack_groups=self.unpack_groups, )
-        space_width = svg_with_space.get_width()
-        return space_width
-
-    def apply_front_and_end_spaces(self):
-        space_width = self.get_space_width()
-        max_height = self.get_height()
-        front_spaces_count = 0
-        i = -1
-        for i in range(self.text.__len__()):
-            if self.text[i] == " ":
-                front_spaces_count += 1
-                continue
-            else:
-                break
-        first_visible_char_index = i
-        if first_visible_char_index != 0:
-            space = Rectangle(width=space_width * front_spaces_count, height=max_height, fill_opacity=0,
-                                  stroke_opacity=0,
-                                  stroke_width=0)
-            text_width = self.get_width()
-            space.move_to(np.array([-text_width / 2, max_height / 2, 0]))
-            self.next_to(space, direction=RIGHT, buff=0)
-            self.submobjects.insert(0, space)
-
-        i = -1
-        last_spaces_count = 0
-        for i in range(self.text.__len__() - 1, -1, -1):
-            if self.text[i] == " ":
-                last_spaces_count += 1
-                continue
-            else:
-                break
-        last_visible_char_index = i
-        if last_visible_char_index != self.text.__len__() - 1:
-            space = Rectangle(width=space_width * last_spaces_count, height=max_height, fill_opacity=0,
-                                  stroke_opacity=0,
-                                  stroke_width=0)
-            text_width = self.get_width()
-            space.move_to(np.array([-text_width / 2, max_height / 2, 0]))
-            self.next_to(space, direction=LEFT, buff=0)
-            self.submobjects.append(space)
-        self.move_to(np.array([0,0,0]))
-
     def apply_space_chars(self):
-        char_index = 0
-        while char_index < self.text.__len__() - 1:
-            char_index += 1
+        for char_index in range(self.text.__len__()):
             if self.text[char_index] == " " or self.text[char_index] == "\t" or self.text[char_index] == "\n":
-                space = Dot(fill_opacity=0, stroke_opacity=0)
-                space.move_to(self.submobjects[char_index - 1].get_center())
+                space = Dot(redius=0, fill_opacity=0, stroke_opacity=0)
+                if char_index == 0:
+                    space.move_to(self.submobjects[char_index - 1].get_center())
```

Comment:
focus on here.
You put all non-display characters in the position of the previous character.
But if `char_index==0`, the previous character is the last display character in the entire Text.

It's better to put them in the position of the first display character instead of the last one.